pages/api/s3-upload.ts

```diff
@@ -1,4 +1,26 @@
-// Folder structure in s3 can be changed in this file if needed
+import { APIRoute } from 'next-s3-upload';
+import { getToken } from 'next-auth/jwt';
+import { NextApiHandler } from 'next';
+
+// Folder structure in s3 can be in the configure if needed
 // see https://next-s3-upload.codingvalue.com/s3-file-paths
+const uploadHandler = APIRoute.configure({});
+const jwtSecret = process.env.NA_JWT_SECRET;
+
+/**
+ * Auth wrapper for next-s3-upload API Route
+ * https://next-s3-upload.codingvalue.com
+ */
+const s3UploadHandler: NextApiHandler = async (req, res) => {
+  const token = await getToken({ req: req, secret: jwtSecret });
+  if (token) {
```

Comment:
Instead of manually using the JWT to check for login, please use `getSession` from NextAuth, unless it's necessary to do this. You can see an example of the use in our GraphQL context configuration: https://github.com/uwblueprint/richmond-centre-for-disability/blob/staging/lib/graphql/context.ts

Reply:
👍 oh nice didn't know we can do that